### PR TITLE
api call

### PR DESCRIPTION
**Pull request by shirrotem**

### CODE REVIEW

File: src/pages/home-page/components/ChefOfTheWeek/ChefOfTheWeek.tsx [hunk 1]
```diff
@@ -3,29 +3,28 @@ import "./ChefOfTheWeek.scss";
...
-    const {dataChefOfTheWeek}= useAppSelector(state=> state.homePage);
-    const {name,img,about,chefRestaurants} = dataChefOfTheWeek;
-    const chefPrivateName= name.split(' ')[0];
-    return (
-        <div className="chefOfTheWeekContainer">
-        <p className="titleChef">CHEF OF THE WEEK:</p>
-        <div className="picAndAboutContainer">
+  const { dataChefOfTheWeek } = useAppSelector((state) => state.homePage);
+  const { name, img, about, chefRestaurants } = dataChefOfTheWeek;
+  const chefPrivateName = name.split(" ")[0];
```
Comment (MoranOshiaMoveo) [inline src/pages/home-page/components/ChefOfTheWeek/ChefOfTheWeek.tsx:9]: Why do u aplit the name?

File: src/pages/home-page/components/HomePage/HomePage.tsx [hunk 2]
```diff
@@ -2,52 +2,68 @@ import React, { useEffect } from "react";
...
+    (state) => state.homePage
+  );
+  useEffect(() => {
+    dispatch(getChefs());
+    dispatch(getPopularRestaurants());
+    dispatch(getSignatureDish());
+    dispatch(getChefOfTheWeek());
+  }, []);
 
+  // console.log(dataRestaurants);
```
Comment (MoranOshiaMoveo) [inline src/pages/home-page/components/HomePage/HomePage.tsx]: Remove comments

File: src/redux-toolkit/slices/homePageSlice.ts [hunk 3]
```diff
@@ -1,44 +1,131 @@
...
+  },
+
+  extraReducers(builder) {
+    builder
+      .addCase(getPopularRestaurants.fulfilled, (state, action: PayloadAction<any>) => {
+        const restaurants = action.payload.map((res: any) => {
+          return {
+            img: res.img,
+            name: res.name,
+            chef: state.dataChefs.find((chef: Chef) => chef.chefRestaurants.includes(res._id)),
```
Comment (MoranOshiaMoveo) [inline src/redux-toolkit/slices/homePageSlice.ts]: The chef should come as a populate and not wait for the call of the chefs

File: src/redux-toolkit/slices/homePageSlice.ts [hunk 4]
```diff
@@ -1,44 +1,131 @@
...
+          };
+        });
+        state.dataDishes = dishes;
+      })
+      .addCase(getSignatureDish.rejected, (state) => {
+        state.dataDishes = [];
+      })
+      .addCase(getChefOfTheWeek.fulfilled, (state, action: PayloadAction<any>) => {
+        const chef = action.payload[0];
+        const chefRes = state.dataRestaurants.filter((res) => chef.restaurants.includes(res._id));
```
Comment (MoranOshiaMoveo) [inline src/redux-toolkit/slices/homePageSlice.ts]: Same in here use populqte

File: src/redux-toolkit/slices/homePageSlice.ts [hunk 5]
```diff
@@ -1,44 +1,131 @@
...
+          chefRestaurants: chefRestaurants,
+          _id: chef._id,
+        };
+      })
+      .addCase(getChefOfTheWeek.rejected, (state) => {
+        state.dataChefOfTheWeek = { img: "", name: "", about: "", chefRestaurants: [], _id: "" };
+      })
+      .addCase(getChefs.fulfilled, (state, action: PayloadAction<any>) => {
+        const chefs = action.payload.map((chef: any) => {
+          const chefRes = state.dataRestaurants.find((res) => chef.restaurants.includes(res._id));
```
Comment (MoranOshiaMoveo) [inline src/redux-toolkit/slices/homePageSlice.ts]: Use populqte

File: src/redux-toolkit/thunks/homePageThunk.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+
+export const getPopularRestaurants = createAsyncThunk("popularRestaurants", async () => {
+  try {
+    const response = await fetch("http://localhost:3000/getPopularRestaurants");
```
Comment (MoranOshiaMoveo) [inline src/redux-toolkit/thunks/homePageThunk.ts]: Set the url as env attr

File: src/data/IconData.tsx [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { Icon } from "./types/icon";
```
Comment (MoranOshiaMoveo) [inline src/data/IconData.tsx:1]: Icons should come from the be

**Review by MoranOshiaMoveo — CHANGES_REQUESTED** [review]

File: src/pages/home-page/components/ChefOfTheWeek/ChefOfTheWeek.tsx (hunk 1 above)
Comment (shirrotem) [reply src/pages/home-page/components/ChefOfTheWeek/ChefOfTheWeek.tsx:9]: To adapt the title to a generic form

File: src/pages/home-page/components/HomePage/HomePage.tsx (hunk 2 above)
Comment (shirrotem) [reply src/pages/home-page/components/HomePage/HomePage.tsx]: fixed

File: src/redux-toolkit/thunks/homePageThunk.ts (hunk 6 above)
Comment (shirrotem) [reply src/redux-toolkit/thunks/homePageThunk.ts]: fixed

File: src/data/IconData.tsx (hunk 7 above)
Comment (shirrotem) [reply src/data/IconData.tsx:1]: fixed

File: src/redux-toolkit/slices/homePageSlice.ts (hunk 3 above)
Comment (shirrotem) [reply src/redux-toolkit/slices/homePageSlice.ts]: fixed

File: src/redux-toolkit/slices/homePageSlice.ts (hunk 4 above)
Comment (shirrotem) [reply src/redux-toolkit/slices/homePageSlice.ts]: fixed

File: src/redux-toolkit/slices/homePageSlice.ts (hunk 5 above)
Comment (shirrotem) [reply src/redux-toolkit/slices/homePageSlice.ts]: fixed

File: src/redux-toolkit/slices/homePageSlice.ts [hunk 8]
```diff
@@ -1,44 +1,111 @@
...
+  dataChefOfTheWeek: Chef;
+  dataChefs: Chef[];
+}
 
+const initialState: HomePageState = {
+  selectedDish: null,
+  isOpen: false,
+  dataRestaurants: [],
+  dataDishes: [],
+  dataChefOfTheWeek: { img: "", name: "", about: "", chefRestaurants: [], _id: "" },
```
Comment (MoranOshiaMoveo) [inline src/redux-toolkit/slices/homePageSlice.ts:20]: Set it as an interface

**Review by MoranOshiaMoveo — APPROVED** [review]

File: src/redux-toolkit/slices/homePageSlice.ts (hunk 8 above)
Comment (shirrotem) [reply src/redux-toolkit/slices/homePageSlice.ts:20]: fixed

**Pull request merged**